File: data/source/tests/pyyal_test_type/open_close-end.py

    # Test open and close.
    ${library_name_suffix}_${type_name}.open(test_source)
    ${library_name_suffix}_${type_name}.close()

    # Test open and close a second time to validate clean up on close.
    ${library_name_suffix}_${type_name}.open(test_source)
    ${library_name_suffix}_${type_name}.close()

    if os.path.isfile(test_source):
      with open(test_source, "rb") as file_object:

        # Test open_file_object and close.
        ${library_name_suffix}_${type_name}.open_file_object(file_object)
        ${library_name_suffix}_${type_name}.close()

        # Test open_file_object and close a second time to validate clean up on close.
        ${library_name_suffix}_${type_name}.open_file_object(file_object)
        ${library_name_suffix}_${type_name}.close()

        # Test open_file_object and close and dereferencing file_object.
        ${library_name_suffix}_${type_name}.open_file_object(file_object)
        del file_object
        ${library_name_suffix}_${type_name}.close()

File: data/source/tests/yal_test_type/open_close-end.c
	/* Test open and close
	 */
#if defined( HAVE_WIDE_SYSTEM_CHARACTER )
	result = ${library_name}_${type_name}_open_wide(
	          ${type_name},
	          source,
	          ${library_name:upper_case}_OPEN_READ,
	          &error );
#else
	result = ${library_name}_${type_name}_open(
	          ${type_name},
	          source,
	          ${library_name:upper_case}_OPEN_READ,
	          &error );
#endif

	${library_name_suffix:upper_case}_TEST_ASSERT_EQUAL_INT(
	 "result",
	 result,
	 1 );

	${library_name_suffix:upper_case}_TEST_ASSERT_IS_NULL(
	 "error",
	 error );

	result = ${library_name}_${type_name}_close(
	          ${type_name},
	          &error );

	${library_name_suffix:upper_case}_TEST_ASSERT_EQUAL_INT(
	 "result",
	 result,
	 0 );

	${library_name_suffix:upper_case}_TEST_ASSERT_IS_NULL(
	 "error",
	 error );

	/* Test open and close a second time to validate clean up on close
	 */
#if defined( HAVE_WIDE_SYSTEM_CHARACTER )
	result = ${library_name}_${type_name}_open_wide(
	          ${type_name},
	          source,
	          ${library_name:upper_case}_OPEN_READ,
	          &error );
#else
	result = ${library_name}_${type_name}_open(
	          ${type_name},
	          source,
	          ${library_name:upper_case}_OPEN_READ,
	          &error );
#endif

	${library_name_suffix:upper_case}_TEST_ASSERT_EQUAL_INT(
	 "result",
	 result,
	 1 );

	${library_name_suffix:upper_case}_TEST_ASSERT_IS_NULL(
	 "error",
	 error );

	result = ${library_name}_${type_name}_close(
	          ${type_name},
	          &error );

	${library_name_suffix:upper_case}_TEST_ASSERT_EQUAL_INT(
	 "result",
	 result,
	 0 );

	${library_name_suffix:upper_case}_TEST_ASSERT_IS_NULL(
	 "error",
	 error );

	/* Clean up
	 */
	result = ${library_name}_${type_name}_free(
	          &${type_name},
	          &error );

	${library_name_suffix:upper_case}_TEST_ASSERT_EQUAL_INT(
	 "result",
	 result,
	 1 );

	${library_name_suffix:upper_case}_TEST_ASSERT_IS_NULL(
	 "${type_name}",
	 ${type_name} );

	${library_name_suffix:upper_case}_TEST_ASSERT_IS_NULL(
	 "error",
	 error );

	return( 1 );

on_error:
	if( error != NULL )
	{
		libcerror_error_free(
		 &error );
	}
	if( ${type_name} != NULL )
	{
		${library_name}_${type_name}_free(
		 &${type_name},
		 NULL );
	}
	return( 0 );
}

